src/mavsdk/core/mavlink_parameters.cpp

```diff
@@ -311,7 +311,19 @@ void MAVLinkParameters::do_work()
                     work->param_value.get_mav_param_ext_type());
             } else {
                 // Param set is intended for Autopilot only.
-                mavlink_msg_param_set_pack(
+                if (_parent.autopilot() == SystemImpl::Autopilot::ArduPilot) {
```

Comment:
Maybe a nitpick, but what about having the `if` just set a temporary variable like this?

```c++
float my_param;
if (_parent.autopilot() == SystemImpl::Autopilot::ArduPilot) {
    my_param = work->param_value.get_4_float_bytes_apm();
} else {
    my_param = work->param_value.get_4_float_bytes();
}
```

This makes it clear that the difference between APM and PX4 is only for this param `my_param` (well, it needs a better name), and all the other parameters given to `mavlink_msg_param_set_pack` are the same.

What do you think?